LibreNMS/Modules/Routes.php

```diff
@@ -0,0 +1,330 @@
+use LibreNMS\Config;
+use LibreNMS\DB\SyncsModels;
+use LibreNMS\Exceptions\InvalidIpException;
+use LibreNMS\Interfaces\Data\DataStorageInterface;
+use LibreNMS\Interfaces\Discovery\RouteDiscovery;
+use LibreNMS\Interfaces\Module;
+use LibreNMS\OS;
+use LibreNMS\Polling\ModuleStatus;
+use LibreNMS\Util\IPv4;
+use LibreNMS\Util\IPv6;
+use SnmpQuery;
+
+class Routes implements Module
+{
+    use SyncsModels;
+
+    /**
+     * @inheritDoc
+     */
+    public function dependencies(): array
+    {
+        return ['ports'];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function shouldDiscover(OS $os, ModuleStatus $status): bool
+    {
+        return $status->isEnabledAndDeviceUp($os->getDevice());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function shouldPoll(OS $os, ModuleStatus $status): bool
+    {
+        return false;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function discover(OS $os): void
+    {
+        $update_timestamp = \Carbon\Carbon::now();
+        $max_routes = (null != Config::get('routes_max_number')) ? Config::get('routes_max_number') : 1000;
+
+        $routes = new Collection;
+
+        if ($os instanceof RouteDiscovery) {
+            $routes = $os->discoverRoutes();
+        }
+
+        if ($routes->isEmpty()) {
+            $routes = $this->discoverInetCidrRoutes($os->getDevice(), $max_routes);
+            $routes = $routes->merge($this->discoverRfcRoutes($os->getDevice()));
+            $routes = $routes->merge($this->discoverIpCidrRoutes($os->getDevice(), $max_routes));
+            $routes = $routes->merge($this->discoverIpv6MibRoutes($os->getDevice()));
+            $routes = $routes->merge($this->discoverVpnVrfRoutes($os->getDevice(), $max_routes));
+        }
+
+        $routes = $routes->filter(function ($data) use ($update_timestamp) {
+            $dst = str_replace('"', '', $data->inetCidrRouteDest ?? '');
+            $hop = str_replace('"', '', $data->inetCidrRouteNextHop ?? '');
+
+            try {
+                preg_match('/(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})$/', (string) $data->inetCidrRoutePfxLen, $tmp); // is it netmask or cidr?
+                $pfxLen = (empty($tmp[1])) ? intval($data->inetCidrRoutePfxLen) : IPv4::netmask2cidr($tmp[1]);
+                Log::debug($dst . ' - ' . $hop . ' - ' . $pfxLen);
+
+                if ($data->inetCidrRouteDestType == 'ipv6') {
+                    $dst = IPv6::fromHexString($dst)->uncompressed();
+                    $hop = IPv6::fromHexString($hop)->uncompressed();
+                } elseif ($data->inetCidrRouteDestType == 'ipv4') {
+                    $tst = new IPv4($dst . '/' . $pfxLen);
+                } else {
+                    return null;
+                }
+            } catch (InvalidIpException $e) {
+                Log::error('Failed to parse IP: ' . $e->getMessage());
+
+                return null;
+            }
+
+            $data->updated_at = $update_timestamp;
```

Comment:
I'm not sure we want to set updated_at, this is going to update every route in the database every discovery, without it, it could possibly skip the update.

Reply:
ok @murrant 
since @PipoCanaja say it is important to keep hystorical routes
and in WEBUI it presented as
![lnms](https://github.com/user-attachments/assets/36277078-9c2a-4760-9f07-aafafd6778ad)

we should keep this as is